src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php

```diff
@@ -1056,60 +1056,73 @@ public function testMessengerTransportConfiguration()
+        $container = $this->createContainerFromFile('messenger_multiple_buses_without_deduplicate_middleware');
 
         $this->assertTrue($container->has('messenger.bus.commands'));
         $this->assertSame([], $container->getDefinition('messenger.bus.commands')->getArgument(0));
-
-        if (class_exists(DeduplicateMiddleware::class)) {
-            $this->assertEquals([
-                ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.commands']],
-                ['id' => 'reject_redelivered_message_middleware'],
-                ['id' => 'dispatch_after_current_bus'],
-                ['id' => 'failed_message_processing_middleware'],
-                ['id' => 'deduplicate_middleware'],
-                ['id' => 'send_message', 'arguments' => [true]],
-                ['id' => 'handle_message', 'arguments' => [false]],
-            ], $container->getParameter('messenger.bus.commands.middleware'));
-        } else {
-            $this->assertEquals([
-                ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.commands']],
-                ['id' => 'reject_redelivered_message_middleware'],
-                ['id' => 'dispatch_after_current_bus'],
-                ['id' => 'failed_message_processing_middleware'],
-                ['id' => 'send_message', 'arguments' => [true]],
-                ['id' => 'handle_message', 'arguments' => [false]],
-            ], $container->getParameter('messenger.bus.commands.middleware'));
-        }
-
+        $this->assertEquals([
+            ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.commands']],
+            ['id' => 'reject_redelivered_message_middleware'],
+            ['id' => 'dispatch_after_current_bus'],
+            ['id' => 'failed_message_processing_middleware'],
+            ['id' => 'send_message', 'arguments' => [true]],
+            ['id' => 'handle_message', 'arguments' => [false]],
+        ], $container->getParameter('messenger.bus.commands.middleware'));
         $this->assertTrue($container->has('messenger.bus.events'));
         $this->assertSame([], $container->getDefinition('messenger.bus.events')->getArgument(0));
+        $this->assertEquals([
+            ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.events']],
+            ['id' => 'reject_redelivered_message_middleware'],
+            ['id' => 'dispatch_after_current_bus'],
+            ['id' => 'failed_message_processing_middleware'],
+            ['id' => 'with_factory', 'arguments' => ['foo', true, ['bar' => 'baz']]],
+            ['id' => 'send_message', 'arguments' => [true]],
+            ['id' => 'handle_message', 'arguments' => [false]],
+        ], $container->getParameter('messenger.bus.events.middleware'));
+        $this->assertTrue($container->has('messenger.bus.queries'));
+        $this->assertSame([], $container->getDefinition('messenger.bus.queries')->getArgument(0));
```

Comment:
Let's convert all `$this->assertSame([], ...)` to `assertEmpty()` while at it maybe?

Reply:
this isn't the same - assertEmpty accepts zero or the empty string, null, etc